src/tribler-gui/tribler_gui/widgets/tablecontentdelegate.py

```diff
@@ -90,10 +90,11 @@ class CheckClickedMixin:
     def check_clicked(self, event, _, __, index):
         model = index.model()
         data_item = model.data_items[index.row()]
+        column_position = model.column_position.get(self.column_name, -1)
         if (
             event.type() == QEvent.MouseButtonRelease
-            and model.column_position.get(self.column_name, -1) == index.column()
-            and data_item[index.model().columns[index.model().column_position[self.column_name]].dict_key] != ''
+            and column_position == index.column()
+            and data_item[model.columns[column_position].dict_key] != ''
```

Comment:
The current version of code (which contains a bug) looks like an attempt to do a logical short-circuiting: if clause A

```python
model.column_position.get(self.column_name, -1) == index.column()
```

is not `True`, then the next clause B

```python
data_item[index.model().columns[index.model().column_position[self.column_name]].dict_key] != ''
```

should not be executed.

Apparently, it fails to do the short-circuiting properly, and clause B is executed even if 

```python
model.column_position.get(self.column_name, -1)
```

is equals to `-1`. In my understanding that means that `index.column()` can be `-1` as well.

The new version of code simplifies the expression, but as it seems to me it still can execute updated clause B2:

```python
data_item[model.columns[column_position].dict_key] != ''
```
when `column_position` is equal to `-1`.

Maybe it is necessary to add another pre-condition `and column_position >= 0` before B2?

Reply:
Strictly speaking, you're right: if `index.column()` equals `-1` the next line will result in taking an incorrect column index. However, `index.column()` can never be `-1` in QT for properly constructed tables.
Anyway, your comment means this is unobvious to a non-QT programmer, so I should add at least a comment.

Reply:
I still think it is possible for `index.column()` to be equal to `-1` in some corner cases. But with your latest change it should be processed correctly 👍